syntax = "proto3";
// This describes the metadata, e.g. the version, the ph number and various features.
message Version_Features {
    string whatsapp_version = 1; // Whatsapp version, for example "2.22.4.14"
    string substringedUserJid = 3; // The last two characters of the user's Jid (phone number)
    // This bools are only written if the backup is a msgstore backup.
    optional bool f_4 = 4; // Booleans that indicate various features
    optional bool f_5 = 5;
    optional bool f_6 = 6;
    optional bool f_7 = 7;
    optional bool f_8 = 8;
    optional bool f_9 = 9;
    optional bool f_10 = 10;
    optional bool f_11 = 11;
    optional bool f_12 = 12;
    optional bool f_13 = 13;
    optional bool f_14 = 14;
    optional bool f_15 = 15;
    optional bool f_16 = 16;
    optional bool f_17 = 17;
    optional bool f_18 = 18;
    optional bool f_19 = 19;
    optional bool f_20 = 20;
    optional bool f_21 = 21;
    optional bool f_22 = 22;
    optional bool f_23 = 23;
    optional bool f_24 = 24;
    optional bool f_25 = 25;
    optional bool f_26 = 26;
    optional bool f_27 = 27;
    optional bool f_28 = 28;
    optional bool f_29 = 29;
    optional bool f_30 = 30;
    optional bool f_31 = 31;
    optional bool f_32 = 32;
    optional bool f_33 = 33;
    optional bool f_34 = 34;
    optional bool f_35 = 35;
    optional bool f_36 = 36;
    optional bool f_37 = 37;
    optional bool f_39 = 39;
}
